src/CodeGeneration.Roslyn.Tool/build/CodeGeneration.Roslyn.Tool.targets

```diff
@@ -24,6 +24,10 @@
         Include="@(Compile)"
         Condition=" '%(Compile.Generator)' == 'MSBuild:GenerateCodeFromAttributes' " />
       <_CodeGenToolInputs_DefineConstants Include="$(DefineConstants)" />
+      <GeneratorBuildProperties Include="@(GeneratorBuildPropertyNames)">
+        <Key>%(Identity)</Key>
+        <Value>$(%(Identity))</Value>
+      </GeneratorBuildProperties>
       <!-- Map GeneratorAssemblySearchPaths to actual DLLs until support is removed -->
```

Comment:
I'd never thought of this syntax to pass arbitrary properties to an msbuild task. It's very clever.

Reply:
About that, I'm wondering whether we shouldn't escape `;` somehow in the Value. If one wanted to pass `TargetFrameworks`, it'd definitely break response file construction and parsing.

Reply:
Good thought. I wouldn't escape it *here* though, since in general an MSBuild item can have arbitrary values for metadata. But when in your msbuild task you create the response file, you should apply whatever escaping is necessary so that the next step will properly parse it.

Reply:
Now escaping semicolons in 6deb257

Reply:
I think @AArnott is right. However I don't know how exactly to do that. Possibly, we need to escape it directly in `_CodeGenToolResponseFileLines` below. At that point we actually use the semicolons to separate lines in response file via including the property into an item type `_CodeGenToolResponseFileContent`.

Reply:
Oooh also, I'm not sure - do we need to Trim `Key` of whitespace? I think we might need to.

Reply:
Originally, I had the transformations all in a single line in the part of the file where the response file lines are created, like this:

@(_CodeGenerationRoslynResolvedProperty->'%(Key)=%(Value)'->Replace(';', '%3B')->'--buildProperty;%(Identity)')

...but I felt that might be a little confusing, so created the 'escaped' item list.  Let me know if there is a better way.  I am also not sure about the necessity of trimming the key, but it seems like a good cautionary measure, so I'll add it in the next commit

Reply:
Now trimming keys in 982c210

Reply:
> do we need to Trim Key of whitespace? I think we might need to.

You only need to if you're abusing properties for what items should be used for. :)

Reply:
So, if we add property content (possibly value like `   a ;   b; ;`) into an item, will it trim and split into a nice
- `a`
- `b`
list?

If it would, we're safe then since we use item for metadata.

Reply:
I think since we're going with items, trimming shouldn't be necessary and escaping Value should happen at the response-file line creation.

Reply:
Ok, removed trimming in e508e37 and moved escaping down to response file line creation in f9cda4b

Reply:
>  if we add property content (possibly value like a ; b; ;) into an item

Property concatenation patterns tend to produce lots of whitespace. One reason why they're *not* good at lists. So ya, if you start with them, you have to clean them up. But if you don't use them at all, the whitespace doesn't tend to happen.